james-sean/server.js

```diff
@@ -0,0 +1,201 @@
+'use strict';
+
+const pg = require('pg');
+const fs = require('fs');
+const express = require('express');
+const bodyParser = require('body-parser');
+const PORT = process.env.PORT || 3000;
+const app = express();
+const conString = 'postgres://localhost:5432/james-sean';
+const client = new pg.Client(conString);
+client.connect();
+client.on('error', function(error) {
+  console.error(error);
+});
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(express.static('./public'));
+
+app.get('/', function(request, response) {
+  response.sendFile('index.html', {root: '.'});
+});
+
+app.get('/new', function(request, response) {
+  response.sendFile('new.html', {root: '.'});
+});
+
+app.get('/articles', function(request, response) {
+  client.query(
+    `SELECT * FROM articles
+     JOIN authors
+     ON articles.author_id = authors.author_id;`
+   )
+  .then(function(result) {
+    response.send(result.rows);
+  })
+  .catch(function(err) {
+    console.error(err)
+  });
+});
+
+app.post('/articles', function(request, response) {
+  client.query(
+    `INSERT INTO authors(author, "authorUrl")
+    VALUES($1, $2) ON CONFLICT DO NOTHING;`,
+    [
+      request.body.author,
+      request.body.authorUrl
+    ]
+  )
+  .then(function() {
+    client.query(
+      `INSERT INTO articles(author_id, title, category, "publishedOn", body)
+      SELECT author_id, $1, $2, $3, $4
+      FROM authors
+      WHERE author=$5;`,
+      [
+        request.body.title,
+        request.body.category,
+        request.body.publishedOn,
+        request.body.body,
+        request.body.author
+      ]
+    )
+  })
+  .then(function() {
+    response.send('Insert complete')
+  })
+  .catch(function(err) {
+    console.error(err)
+  });
+});
+
+app.put('/articles/:id', function(request, response) {
+  client.query(
+    `UPDATE authors
+    SET author=$1, "authorUrl"=$2
+    WHERE author_id=$3;`,
+    [
+      request.body.author,
+      request.body.authorUrl,
+      request.params.id
+    ]
+  )
+  .then(function() {
+    client.query(
+      `UPDATE articles
+      SET title=$1, category=$2, "publishedOn"=$3, body=$4
+      WHERE article_id=$5;`,
+      [
```

Comment:
Missing the author ID here as part of the request body. Make sure to always look closely at the original request in the model while making sql requests.